source/native-plugins/midi-transpose.c

```diff
@@ -50,57 +52,85 @@ static void miditranspose_cleanup(NativePluginHandle handle)
 
 static uint32_t miditranspose_get_parameter_count(NativePluginHandle handle)
 {
-    return 1;
+    return 2;
 
     // unused
     (void)handle;
 }
 
 static const NativeParameter* miditranspose_get_parameter_info(NativePluginHandle handle, uint32_t index)
 {
-    if (index != 0)
+    if (index >= miditranspose_get_parameter_count(handle))
         return NULL;
 
-    static NativeParameter param;
-
-    param.name  = "Octaves";
-    param.unit  = NULL;
-    param.hints = NATIVE_PARAMETER_IS_ENABLED|NATIVE_PARAMETER_IS_AUTOMABLE|NATIVE_PARAMETER_IS_INTEGER;
-    param.ranges.def = 0.0f;
-    param.ranges.min = -8.0f;
-    param.ranges.max = 8.0f;
-    param.ranges.step = 1.0f;
-    param.ranges.stepSmall = 1.0f;
-    param.ranges.stepLarge = 1.0f;
-    param.scalePointCount = 0;
-    param.scalePoints     = NULL;
-
-    return &param;
-
-    // unused
-    (void)handle;
+    static NativeParameter param[] =
+    {
+        {
+          .name  = "Octaves",
+          .unit  = NULL,
+          .hints = NATIVE_PARAMETER_IS_ENABLED|NATIVE_PARAMETER_IS_AUTOMABLE|NATIVE_PARAMETER_IS_INTEGER,
+          .ranges.def = 0.0f,
+          .ranges.min = -8.0f,
+          .ranges.max = 8.0f,
+          .ranges.step = 1.0f,
+          .ranges.stepSmall = 1.0f,
+          .ranges.stepLarge = 1.0f,
+          .scalePointCount = 0,
+          .scalePoints     = NULL,
+        },
+        {
+          .name  = "Semitones",
+          .unit  = NULL,
+          .hints = NATIVE_PARAMETER_IS_ENABLED|NATIVE_PARAMETER_IS_AUTOMABLE|NATIVE_PARAMETER_IS_INTEGER,
+          .ranges.def = 0.0f,
+          .ranges.min = -12.0f,
+          .ranges.max = 12.0f,
+          .ranges.step = 1.0f,
+          .ranges.stepSmall = 1.0f,
+          .ranges.stepLarge = 1.0f,
+          .scalePointCount = 0,
+          .scalePoints     = NULL,
+        },
+    };
+    return &param[index];
 }
 
 static float miditranspose_get_parameter_value(NativePluginHandle handle, uint32_t index)
 {
-    if (index != 0)
-        return 0.0f;
-
-    return (float)handlePtr->octaves;
+    switch (index)
+    {
+        case 0:
+            return (float)handlePtr->octaves;
+            break;
```

Comment:
please remove this `break` it does nothing here as the return completes the execution of the function.
same with the one below it.